prow.go

```diff
@@ -955,12 +1033,17 @@ func findTargetName(spec *corev1.PodSpec) (string, error) {
 					if len(value) == 0 {
 						return "", fmt.Errorf("bug in resolving %s", value)
 					}
-					return value, nil
+					targetName = value
 				}
 			}
 		}
 	}
-	return "", fmt.Errorf("could not find argument --target=X in prow job pod spec to identify target pod name")
+
+	if len(targetName) > 0 {
+		return targetName, nil
+	} else {
```

Comment:
don't else when the if returns

Reply:
but return errors early

Reply:
All changes to the method reverted. They were no longer necessary with the change to determining step mode.